### PR TITLE
fix: Include string literal object keys as used

### PR DESCRIPTION
**Pull request by eps1lon**
String literals as object keys are currently ignored. These are used to destructure e.g. aria-* props or data-* props:
```jsx
export default function Dialog(props) {
	const { 'aria-describedby': ariaDescribedby } = props;
	return null;
}
```

### CODE REVIEW

File: src/generator.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export function generate(node: t.Node | t.PropTypeNode[], options: GenerateOptio
 
 		return filteredNodes
 			.map((prop) => generate(prop, options))
-			.reduce((prev, curr) => `${prev}\n${curr}`);
+			.reduce((prev, curr) => `${prev}\n${curr}`, '');
```
Comment (eps1lon) [inline src/generator.ts]: This is isn't needed if the PRs is merged. It prevents a crash if no used props were found. Not sure if we should throw or warn but it should be something other than "TypeError: called reduce on empty array without initial value".

File: jest.config.js [hunk 2]
```diff
@@ -4,10 +4,10 @@ const path = require('path');
 module.exports = {
 	testEnvironment: 'node',
 	transform: defaults.transform,
-	testRegex: `test/index.test.ts`,
+	testRegex: `test/.*\.test\.ts`,
```
Comment (eps1lon) [inline jest.config.js]: needs more work. currently this considers snapshots as test files.

Comment (eps1lon) [reply jest.config.js]: fixed

**Review by merceyz — CHANGES_REQUESTED** [review]
Interesting that material-ui didn't come across this before, thanks for fixing it!

I finally got around to finish the testing architecture I started on some time ago, could you update the test to use it please

File: src/generator.ts (hunk 1 above)
Comment (merceyz) [reply src/generator.ts]: I agree that it should be handled better than what happens right now, but you can revert it for this PR.

Comment (eps1lon) [reply src/generator.ts]: done

**Review by merceyz — APPROVED** [review]
Updated the test to use the fixtures instead, otherwise looks good. Thanks

File: src/generator.ts (hunk 1 above)
Comment (merceyz) [reply src/generator.ts]: Fixed in https://github.com/merceyz/typescript-to-proptypes/commit/b69112e1011f089b6d5cb60f88ce75b6394252be

**Pull request merged**